staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -660,11 +659,16 @@ func updateURLMetrics(req *Request, resp *http.Response, err error) {
 // Any non-2xx http status code causes an error.  If we get a non-2xx code, we try to convert the body into an APIStatus object.
 // If we can, we return that as an error.  Otherwise, we create an error that lists the http status and the content of the response.
 func (r *Request) Stream() (io.ReadCloser, error) {
+	ctx := context.Background()
```

Comment:
nit, flip this to:
```go
ctx := r.ctx
if ctx == nil {
  ctx = context.Background()
}
```

avoids an allocation, and later when ctx becomes a parameter to this method, we can just drop the `ctx := r.ctx` line

Reply:
oh, unless we require ctx to be non-nil in all invocations... in which case this is temporary/fine

Reply:
Yep. This should be temporary.